README.pt-br.md

```diff
@@ -1296,38 +1306,132 @@ get '/:value' do
+dos casos um valor aleatório criptograficamente seguro de um comprimento
+apropriado que para `HMAC-SHA1` é maior que ou igual a 64 bytes (512 bits, 128
+caracteres hexadecimais). Você é aconselhado a não usar um segredo menor que 32
+bytes de aleatoriedade (256 bits, 64 caracteres hexadecimais). Por isso é
+**muito importante** que você não faça o segredo na mão, mas em vez disso use um
+gerador de números aleatórios seguro para criá-lo. Humanos são extramemente
+ruins em gerar valores aleatórios.
 
-get '/' do
-  "value = " << session[:value].inspect
-end
+Por padrão, um segredo de sessão seguro de 32 bytes é gerado pra você pelo
+Sinatra, mas ele será mudado a cada reinicialização da sua aplicação. Se sua
+aplicação tiver múltiplas instâncias, e deixar o Sinatra gerar a chave, cada
+instância terá uma chave de sessão diferente, algo que você provavelmente não
+vai querer.
 
-get '/:value' do
-  session['value'] = params['value']
-end
+Para mais segurança e usabilidade é [recomendado](https://12factor.net/config)
+que você gere um segredo aleatório seguro e guarde-o em uma variável de ambiente
+em cada host que esteja executando sua aplicação, assim todas as suas instâncias
+da aplicação vão compartilhar o mesmo segredo. Você deve periodicamente
+rotacionar este segredo de sessão por um valor novo. Aqui estão alguns exemplos
+de como você pode criar um segredo de 64 bytes e usá-lo:
+
+**Gerando o Segredo da Sessão**
+
+```text
+$ ruby -e "require 'securerandom'; puts SecureRandom.hex(64)"
+99ae8af...snip...ec0f262ac
 ```
 
-Para melhorar a segurança, os dados da sessão guardados no cookie é assinado com uma chave secreta da sessão. Uma chave aleatória é gerada para você pelo Sinatra. Contudo, já que a chave mudará cada vez que você inicia sua aplicação, você talvez queira defini-la você mesmo para que todas as instâncias da aplicação compartilhe-a:
+**Gerando o Segredo da Sessão (Bônus)**
+
+Use o [sysrandomgem](https://github.com/cryptosphere/sysrandom#readme) para usar
+as facilidades do sistema RNG de gerar valores aleatórios em vez do userspace
+`OpenSSL` que é o padrão atual do MRI Ruby:
+
+```text
+$ gem install sysrandom
+Building native extensions.  This could take a while...
+Successfully installed sysrandom-1.x
+1 gem installed
+
+$ ruby -e "require 'sysrandom/securerandom'; puts SecureRandom.hex(64)"
+99ae8af...snip...ec0f262ac
+```
+
+**Variável de Ambiente do Segredo da Sessão**
+
+Crie uma variável de ambiente `SESSION_SECRET` para o Sinatra com o valor que
+foi gerado por você. Faça com que esse valor fique persistente através das
+reinicializações do seu host. O método para fazer isso vai variar de sistema
+para sistema, este é apenas para propósitos ilustrativos:
+
+```bash
+# echo "export SESSION_SECRET=99ae8af...snip...ec0f262ac" >> ~/.bashrc
+```
+
+**Session Secret App Config**
+
+Configure sua aplicação para torná-la a prova de falhas caso o segredo aleatório
```

Comment:
Hm, I think what they meant in the original version was that in case the variable was not set at all, you need to have a code fallback in place. The impression I get here is that the variable is there, but empty.

```suggestion
Configure sua aplicação para ter um fallback com um segredo aleatório seguro
```